src/pthread-main.js

```diff
@@ -26,31 +29,32 @@ var Module = {};
 
 function threadPrint() {
   var text = Array.prototype.slice.call(arguments).join(' ');
-  postMessage({cmd: 'print', text: text, threadId: selfThreadId});
+  console.log(text);
 }
 function threadPrintErr() {
   var text = Array.prototype.slice.call(arguments).join(' ');
-  postMessage({cmd: 'printErr', text: text, threadId: selfThreadId});
+  console.error(text);
 }
 function threadAlert() {
   var text = Array.prototype.slice.call(arguments).join(' ');
   postMessage({cmd: 'alert', text: text, threadId: selfThreadId});
 }
 Module['print'] = threadPrint;
 Module['printErr'] = threadPrintErr;
-
-// Work around https://bugzilla.mozilla.org/show_bug.cgi?id=1049091
-console = {
-  log: threadPrint,
-  error: threadPrintErr
-};
-
 this.alert = threadAlert;
 
 this.onmessage = function(e) {
   if (e.data.cmd === 'load') { // Preload command that is called once per worker to parse and load the Emscripten code.
-    buffer = e.data.buffer;
+    // Initialize the thread-local field(s):
     tempDoublePtr = e.data.tempDoublePtr;
+
+    // Initialize the global "process"-wide fields:
+    buffer = e.data.buffer;
+    TOTAL_MEMORY = e.data.TOTAL_MEMORY;
+    STATICTOP = e.data.STATICTOP;
+    DYNAMIC_BASE = e.data.DYNAMIC_BASE;
+    DYNAMICTOP_PTR = e.data.DYNAMICTOP_PTR;
+
```

Comment:
Here we now explicitly receive the global system fields from the main thread, instead of recomputing them.